static/assets/js/entity-reference/app.js

```diff
@@ -53,30 +63,44 @@
                             return entity.name == name;
                         });
                     }
-                    return entity;
+                    return !entity.details['x-hidden'];
                 })
-                .map(setDisplayTypesOnEntity);
+                .map(function(entity) 
+                { 
+                    return setDisplayTypesOnEntity(entity, swagger.definitions); 
+                });
         }
 
-        function setDisplayTypesOnEntity(entity) {
+        function setDisplayTypesOnEntity(entity, definitions) {
             Object.keys(entity.details.properties).forEach(function(propertyName) {
                 var property = entity.details.properties[propertyName];
+                property.isArray = (property.type === "array");
 
-                if (property.type == "array") {
-                    var ref = property.items.$ref.replace("#/definitions/", "");
-                    property.displayType = ref;
-                    property.ref = ref;
-                    property.isArrayRef = true;
-                } else if (property.$ref != undefined) {
-                    var ref = property.$ref.replace("#/definitions/", "");
-                    property.displayType = ref;
-                    property.ref = ref;
-                    property.isSingleRef = true;
+                if (property.isArray) {
+                    property.ref = (property.items.$ref && property.items.$ref.replace("#/definitions/", ""));
+                    property.displayType = getRefDisplayName(property.ref, definitions) || getTypeDisplayName(property.items);
                 } else {
-                    property.displayType = property.type;
-                }
+                    property.ref = property.$ref && property.$ref.replace("#/definitions/", "");
+                    property.displayType = getRefDisplayName(property.ref, definitions) || getTypeDisplayName(property);
+                } 
+
+                property.descriptionHtml = $sce.trustAsHtml(property.description);
             });
             return entity;
         }
+
+        function getRefDisplayName(ref, definitions) {
+            if (ref) {
+                return (definitions[ref]['x-display-name'] || ref).toLowerCase();
+            }
+        }
+
+        function getTypeDisplayName(property) {
```

Comment:
I was viewing it more as being a single case now. When we have multiple cases, then we can refactor it (if need be). Also, the code would all be contained in a single easy to grok function. It isn't like the current one is NOT easy to grok. We just have a bit of eye moving / scroll to see the table.

I think it is fine the way it is, if you do. I don't think the performance is a compelling argument, but ease of adding new mappings is a good argument.